static/app/views/issueDetails/metricIssues/utils.tsx

```diff
@@ -73,5 +77,73 @@ export function useMetricTimePeriod({
       ),
       custom: true,
     };
-  }, [openPeriod]);
+  }, [openPeriod, since, until]);
+}
+
+export function useMetricIssueLegend({
+  rule,
+  series,
+}: {
+  rule: MetricRule;
+  series: Series[];
+}) {
+  const theme = useTheme();
+  return useMemo(() => {
+    const legendSet = new Set(series.map(s => s.seriesName));
+    if (legendSet.has(rule.aggregate)) {
+      legendSet.delete(rule.aggregate);
+    }
+    const legendItems = Array.from(legendSet).map(name => {
+      switch (name) {
+        case 'Threshold Line':
+          return {
+            name,
+            icon: HORIZONTAL_DASHED_LINE_CHART_ICON,
+            itemStyle: {
+              color: theme.error,
+            },
+          };
+        case 'Incident Line':
+          return {
+            name,
+            icon: VERTICAL_LINE_CHART_ICON,
+            itemStyle: {
+              color: theme.error,
+            },
+          };
+
+        case 'Status Area':
+          return {
+            name,
+            icon: HORIZONTAL_WIDE_LINE_CHART_ICON,
+            itemStyle: {
+              color: theme.error,
+            },
+          };
+        default:
+          return {
+            name,
+            icon: 'circle',
+            color: theme.gray300,
+          };
+      }
+    });
+    return Legend({
+      theme,
+      orient: 'horizontal',
+      align: 'left',
+      show: true,
+      data: legendItems,
+      inactiveColor: theme.gray200,
+    });
+  }, [rule, series, theme]);
 }
+
+const HORIZONTAL_DASHED_LINE_CHART_ICON =
```

Comment:
these potentially seem generic enough that we can put them with the either icons and have them in our component library? or adding a comment to this file? it mights be a little confusing for someone to look at this file and see these 3 svg paths without much context

Reply:
Added a comment, but these can't really be used anywhere else since they aren't actually SVG elements. They can only be used with the `legend.icon` field for eCharts since they match the formatting: https://echarts.apache.org/en/option.html#legend.icon. I added comments in case anyone happens upon these, but elsewhere in the app these are not toggle-able so this is kind of a first.